docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+ - `checkClash(ArrayList< item >, Item)`- Updates `isClash` once a duplicate item is found in the list.
+ - `checkRecurrenceDate(Task)` - Checks if the current date is beyond the stated date in the list, and provides a new update for the date recurring date.
+
+Given below is an example usage scenario and how the checker mechanism behaves at each step. 
+
+![here](Images/Checker_Diagram.png)
+
+Step 1. A new `Deadline` object is created and needs to be added to the existing list of task. Hence it calls `addTask()` method under `Data` class. 
+
+Step 2. Data instantiates a Checker with its existing list by calling its constructor, and the task to be added to the list 
+
+Step 3. Data proceeds to call the `checkRecurrenceDate(Task)` of the Checker class, to get a newDate if today's date is beyond the stated weekly date.
+ 
+Step 4. If `newDate` is not `null`it shows that there is a new updated date. Therefore, we proceed to update the object with our updated weekly date. 
+
+Step 5. Now we proceed to call `checkDuplicates()` of Checker class. 
+
+Step 6. If `false` , there is no duplicates in the existing list, and the task can be safely added. Otherwise, no action will be taken. 
+
+**Design consideration: 
+Aspect: How checker executes**
+
+ - **Alternative 1(current choice):** Check for clashes *before* adding task onto list: 
+	 - Pros: Easy to implement as we know specifically what to find in the list eg similar dates & description.
+	 - Cons: Delays the efficiency of adding tasks onto list. 
+ - **Alternative 2:** Check for clash after task is being added onto list 
+	 - Pros: Does not hinder the speed of task adding. 
+	 - Cons: Harder to implement as we have to loop through the entire list to look for duplicates. 
+
+### 4.4 CAP calculator feature
+
+This feature extends `Action` to execute command given by the user, output are then passed on to `Ui` for display. 
+Additionally, it implements the following operations:
+
+* `CalculateCapAction#act()` - Calculate the user CAP based on stored user grades / input modules.
+* `CalculateCapAction#prepare()` - Parse user command to suitable parameter for `CalculateCapAction#act()` function.
+
+Given below is an example usage scenario and how thecap calculator mechanism behaves at each step.
+
+Step 1. The user executes `cap` command find his current CAP grade. Command is then parsed by `CalculateCapAction#prepare()` to be passed as arguments for `CalculateCapAction#act()`.
+
+Step 2. `CalculateCapAction#act()` retrieves data from the stored user's grades.
+
+Step 3. `CalculateCapAction#act()` then retrieves module data from the `modulelist.txt` to determine Modular Credit (MC) allocation.
+
+Step 4. CAP value is calculated and returned to the user through `Ui`.
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![cap uml diagram](Images/CalculateCapSequence.png)
```

Comment:
Same problem encountered as the above reviewers causing the diagram to be messy. Maybe there was a problem with the picture file.